extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -360,4 +354,36 @@ private QueryDefinitionBuilder makeQueryDefinitionBuilder(String queryId, DataSo
     }
     return queryDefBuilder;
   }
+
+  /**
+   * Computes the ClusterBy for the final window stage which may or may not have the partition boosted column,
+   * depending on the {@code segmentGranularity} parameter passed. We don't have to take the CLUSTERED BY
+   * columns into account, as they are handled as {@link org.apache.druid.query.scan.ScanQuery#orderBys}.
+   */
+  private static ClusterBy computeClusterByForFinalWindowStage(Granularity segmentGranularity)
```

Comment:
Partition boosting should be done irrespective of the granularity IMO.

Reply:
Why though?

Reply:
What's the harm?

Reply:
It's an unnecessary column, so we shouldn't add it unless we need to? Same reason why we are not adding it to all the window stages.

Reply:
Hmm, I see your point.

Why is partition boosting conditional in GroupByQueryKit then?

Reply:
Currently, the boost column is a workaround for when the clustered by is empty (i.e. granularity = ALL). It is also an optimization that breaks up large partitions. Hence even if the query doesn't throw when granularity =/= ALL, it still is worth it to have that optimisation for other granularities as well.

Reply:
@LakshSingla IIUC, do you mean it should be unconditional in GroupByQueryKit as well?

Reply:
GroupBy is different due to the fact that it aggregates as well, therefore the chance of having a large partition if the partition is on the grouping dimensions is negligible (haven't seen anything like that in practice). Therefore there can't be large partitions when the post shuffle spec = pre shuffle spec. There can be large partitions if that isn't true which is when we have group by + order by etc. Window, scan kits don't aggregate, therefore they have to worry about partition sizes.

Reply:
@LakshSingla I see. Have made the suggested change to add partition boosting irrespective of segment granularity. Thanks for pointing this out!